client/service/src/client/call_executor.rs

```diff
@@ -101,6 +111,16 @@ where
 		)?;
 		let state = self.backend.state_at(*id)?;
 		let state_runtime_code = sp_state_machine::backend::BackendRuntimeCode::new(&state);
+		let runtime_code = {
+			let code = state_runtime_code.runtime_code()?;
+			let spec = self.runtime_version(id)?.spec_version;
```

Comment:
Same problem.

And now this code exists 3 times. Could be moved in a function?

Reply:
Encapsulated code into its own function `check_overwrite`. This function is sortof similar to the original `try_replace`except native to `CallExecutor`. The function accepts a `RuntimeCode` which couldn't be moved into the function because of the lifetime/reference present on `RuntimeCode<'a>`